File: examples/users-and-groups/group-membership.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Add and list group members; list a user's group memberships.
export async function manageGroupMembers(client: JsBaoClient, userId: string) {
  // #region example
  // Add a member by email (recommended for user-facing flows)
  const result = await client.groups.addMember("team", "engineering", {
    email: "alice@example.com",
  });

  // ...or by user id (internal / programmatic)
  await client.groups.addMember("team", "engineering", { userId: "user-456" });

  // List a group's members
  const members = await client.groups.listMembers("team", "engineering");

  // List the groups a user belongs to
  const memberships = await client.groups.listUserMemberships(userId);
  // #endregion example
  return { result, members, memberships };
}

File: examples/users-and-groups/group-membership.swift
import JsBaoClient

// Add and list group members; list a user's group memberships.
func manageGroupMembers(client: JsBaoClient, userId: String) async throws {
  // #region example
  // Add a member by email (recommended for user-facing flows)
  let result = try await client.groups.addMember(
    groupType: "team", groupId: "engineering",
    params: ["email": "alice@example.com"]
  )

  // ...or by user id (internal / programmatic)
  _ = try await client.groups.addMember(
    groupType: "team", groupId: "engineering",
    params: ["userId": "user-456"]
  )

  // List a group's members
  let members = try await client.groups.listMembers(groupType: "team", groupId: "engineering")

  // List the groups a user belongs to
  let memberships = try await client.groups.listUserMemberships(userId: userId)
  // #endregion example
  _ = (result, members, memberships)
}
